packages/devtools_app/lib/src/app_size/app_size_screen.dart

```diff
@@ -93,6 +99,43 @@ class _AppSizeBodyState extends State<AppSizeBody>
     addAutoDisposeListener(_tabController);
   }
 
+  Future<void> maybeLoadAppSizeFiles() async {
+    final queryParams = loadQueryParams();
+    if (queryParams.containsKey(primaryAppSizeFilePropertyName)) {
+      preLoadingData = true;
+      final primaryAppSizeFile = await server.requestPrimaryAppSizeFile(
+          queryParams[primaryAppSizeFilePropertyName]);
+      DevToolsJsonFile secondaryAppSizeFile;
+      if (queryParams.containsKey(secondaryAppSizeFilePropertyName)) {
+        secondaryAppSizeFile = await server.requestSecondaryAppSizeFile(
+            queryParams[secondaryAppSizeFilePropertyName]);
+      }
+
+      if (primaryAppSizeFile != null) {
```

Comment:
should we throw an exception `if primaryAppSizeFile == null`

Reply:
added a todo to add better error handling. need to get some of this code landed to build implementation in google3.